lib/workers/repository/update/branch/get-updated.ts

```diff
@@ -526,3 +548,168 @@ async function applyManagerBumpPackageVersion(
 
   return result.bumpedContent;
 }
+
+function updateBranchConfigAfterArtifacts(config: BranchConfig): void {
+  //config.schedule = [];
+  config.commitBody = '[SECURITY] - This update addresses a vulnerability.';
+  config.prCreation = 'immediate';
+  config.prHeader =
+    'This PR was generated to address a detected security vulnerability.';
+}
+
+function isFileAddition(file: unknown): file is FileAddition {
+  return (
+    !!file && typeof file === 'object' && (file as any).type === 'addition'
+  );
+}
+
+function parseRpmtestArtifactsResults(results: UpdateArtifactsResult[] | null):
+  | {
+      name: string;
+      oldVersion: string;
+      newVersion: string;
+    }[]
+  | null {
+  if (!results) {
+    return null;
+  }
+  const rpmRegex = /^(?<name>.+)-(?<version>[^-]+-[^-]+)\.src\.rpm$/;
+  const changes = results.flatMap((res) => {
+    let parsedContents = null;
+    let parsedPreviousContents = null;
+    const rpmVersionChanges: {
+      name: string;
+      oldVersion: string;
+      newVersion: string;
+    }[] = [];
+    if (isFileAddition(res.file)) {
+      try {
+        if (res.file.contents) {
+          parsedContents = yaml.load(res.file.contents.toString());
+        }
+      } catch (e) {
+        parsedContents = {
+          error: 'Failed to parse contents as YAML',
+          details: e,
+        };
+      }
+      try {
+        if (res.file.previousContents) {
+          parsedPreviousContents = yaml.load(
+            res.file.previousContents.toString(),
+          );
+        }
+      } catch (e) {
+        parsedPreviousContents = {
+          error: 'Failed to parse previousContents as YAML',
+          details: e,
+        };
+      }
+      // Extract sourcerpm fields from both YAMLs
+      const getSourcerpms = (parsed: any): string[] => {
+        if (!parsed) {
+          return [];
+        }
+        const rpms: string[] = [];
+        const search = (obj: any): void => {
+          if (Array.isArray(obj)) {
+            obj.forEach(search);
+          } else if (obj && typeof obj === 'object') {
+            for (const [key, value] of Object.entries(obj)) {
+              if (key === 'sourcerpm' && typeof value === 'string') {
+                rpms.push(value);
+              } else {
+                search(value);
+              }
+            }
+          }
+        };
+        search(parsed);
+        return rpms;
+      };
+      const newSourcerpms = getSourcerpms(parsedContents);
+      const oldSourcerpms = getSourcerpms(parsedPreviousContents);
+      // Map by name for easy comparison
```

Comment:
I think this will suffer from the same problem @qixiang talked about, that the NEVRA can come from multiple repos. But adding the source repo URL here might be possible compared to other places where we have tried.